lib/custom_a11y_rules.js

```diff
@@ -0,0 +1,150 @@
+var customRules = {
+  "rules": [
+    {
+      "id": "nav-aria-label",
+      "selector": "nav",
+      "enabled": true,
+      "tags": [ "edx-custom" ],
+      "all": [
+        "nav-aria-label-present",
+        "nav-aria-label-value"
+      ],
+      "any": [],
+      "none": [],
+      "metadata": {
+        "description": "Ensures <nav> elements have valid aria-label attribute.",
+        "help": "<nav> elements MUST have an aria-label, and the aria-label MUST NOT have 'navigation' or 'nav' as part of the value.",
+        "helpUrl": "https://openedx.atlassian.net/wiki/display/A11Y/edX+Specific+Accessibility+Tests"
+      }
+    },
+    {
+      "id": "link-href",
+      "selector": "a",
+      "enabled": true,
+      "tags": [ "edx-custom" ],
+      "all": [
+        "link-href",
+        "link-href-internal"
+      ],
+      "any": [],
+      "none": [],
+      "metadata": {
+        "description": "Ensures <a> elements have acceptable href values",
+        "help": "all links must have an href whose value MUST NOT be empty or \"#\"",
+        "helpUrl": "https://openedx.atlassian.net/wiki/display/A11Y/edX+Specific+Accessibility+Tests"
+      }
+    },
+    {
+      "id": "skip-link",
+      "selector": "a[href]",
+      "pageLevel": true,
+      "enabled": true,
+      "tags": ["edx-custom"],
+      "all": [],
+      "any": ["skip-link"],
+      "none": [],
+      "metadata": {
+          "description": "Ensures the first link on the page is a skip link",
+          "help": "The page should have a skip link as its first link",
+          "helpUrl": "https://dequeuniversity.com/rules/axe/1.1/skip-link"
+      }
+    },
+    {
+      "id": "icon-aria-hidden",
+      "selector": "i.icon, i.fa, span.icon, span.fa",
```

Comment:
@benpatterson  RE: "I don't see the 'i' element being selected like how similar elements are selected in above tests...how is that done for this function?"

This is the `icon-aria-hidden` "Rule" which uses the `icon-aria-hidden` "Check".  The `selector` attribute of the rule is where you can define which elements to run the checks on. Elements that match are passed as `node` to the checks' `evaluate` method.  In edX, these are any `<i>` or `<span>` elements with the "fa" or "icon" class.